src/macaron/repo_finder/repo_finder.py

```diff
@@ -218,3 +213,158 @@ def find_source(purl_string: str, repo: str | None) -> bool:
     logger.info("%s/commit/%s", found_repo, digest)
 
     return True
+
+
+def prepare_repo(
```

Comment:
Thanks for making the refactoring. However, I think `prepare_repo`, `get_git_service` and `get_local_repos_path` can be moved to a new package within `src/macaron/` (for example, `prepare_repo` is used both within `analyzer.py` and `repo_finder.py`).
We could name this new package something like "repo_utils" or something like that 🤔 
@behnazh-w I would love to know what you think about this.